Tests/TestingTests/MiscellaneousTests.swift

```diff
@@ -615,43 +615,45 @@ struct MiscellaneousTests {
       0xABCD1234,
       0,
       { outValue, hint in
-        if let hint, hint.loadUnaligned(as: TestContentAccessorHint.self) != expectedHint {
+        if let hint, hint.load(as: TestContentAccessorHint.self) != expectedHint {
           return false
         }
         _ = outValue.initializeMemory(as: TestContentAccessorResult.self, to: expectedResult)
         return true
       },
-      UInt(UInt64(0x0204060801030507) & UInt64(UInt.max)),
+      UInt(truncatingIfNeeded: UInt64(0x0204060801030507)),
       0
     )
   }
 
   @Test func testDiscovery() async {
-    await confirmation("Can find a single test record") { found in
-      DiscoverableTestContent.enumerateTestContent { _, value, context, _ in
-        if value == DiscoverableTestContent.expectedResult && context == DiscoverableTestContent.expectedContext {
-          found()
-        }
-      }
-    }
-
-    await confirmation("Can find a test record with matching hint") { found in
+    // Check the type of the test record sequence (it should be lazy.)
+    let allRecords = DiscoverableTestContent.discover()
```

Comment:
Not strictly related to this PR, but it's occurring to me seeing this that it might make sense to migrate these tests (and their fixtures) related to test discovery into a distinct new test file and suite.

Reply:
I had the same thought, but I am going to wait until the test content emission PR has landed and there are more tests we can think of to write.